src/screens/UserManagement/CreateCustomRole.res

```diff
@@ -0,0 +1,153 @@
+        />
+      </div>
+      <div className="absolute top-10 right-5">
+        <FormRenderer.SubmitButton text="Create role" loadingText="Loading..." />
+      </div>
+    </div>
+  }
+}
+
+@react.component
+let make = (~isInviteUserFlow=true, ~setNewRoleSelected=_ => ()) => {
+  open APIUtils
+  open LogicUtils
+  open UIUtils
+  let fetchDetails = useGetMethod()
+  let updateDetails = useUpdateMethod()
+
+  let initialValuesForForm = [("role_scope", "merchant"->JSON.Encode.string)]->Dict.fromArray
+
+  let {permissionInfo, setPermissionInfo} = React.useContext(GlobalProvider.defaultContext)
+  let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Loading)
+
+  let paddingClass = isInviteUserFlow ? "p-10" : ""
+  let marginClass = isInviteUserFlow ? "mt-5" : ""
+
+  let (groupsAdded, setGroupsAdded) = React.useState(_ => [])
+
+  let onSubmit = async (values, _) => {
+    try {
+      setScreenState(_ => PageLoaderWrapper.Loading)
+      let url = getURL(~entityName=USERS, ~userType=#CREATE_CUSTOM_ROLE, ~methodType=Post, ())
+      let body = values->LogicUtils.getDictFromJsonObject
+      let arrayVal = groupsAdded->Array.map(JSON.Encode.string)
+      body->Dict.set("groups", arrayVal->JSON.Encode.array)
+      let _ = await updateDetails(url, body->JSON.Encode.object, Post, ())
+      setScreenState(_ => PageLoaderWrapper.Success)
+    } catch {
+    | _ => ()
+    }
+    Nullable.null
+  }
+
+  let getPermissionInfo = async () => {
+    try {
+      setScreenState(_ => PageLoaderWrapper.Loading)
+      let url = getURL(~entityName=USERS, ~userType=#PERMISSION_INFO, ~methodType=Get, ())
+      let res = await fetchDetails(`${url}?groups=true`)
+      let permissionInfoValue = res->getArrayDataFromJson(ProviderHelper.itemToObjMapperForGetInfo)
+      setPermissionInfo(_ => permissionInfoValue)
+      setScreenState(_ => PageLoaderWrapper.Success)
+    } catch {
+    | _ => setScreenState(_ => PageLoaderWrapper.Error("Something went wrong!"))
+    }
+  }
+
+  React.useEffect0(() => {
+    if permissionInfo->Array.length === 0 {
+      getPermissionInfo()->ignore
+    }
+    None
+  })
+
+  <div className="flex flex-col overflow-y-scroll h-full">
+    <RenderIf condition={isInviteUserFlow}>
+      <BreadCrumbNavigation
+        path=[{title: "Users", link: "/users"}] currentPageTitle="Create custom roles"
+      />
+      <PageUtils.PageHeading
+        title="Create custom roles"
+        subTitle="An invite will be sent to the email addresses to set up a new account"
+      />
+    </RenderIf>
+    <div
+      className={`h-4/5 bg-white relative overflow-y-scroll flex flex-col gap-10 ${paddingClass} ${marginClass}`}>
+      <PageLoaderWrapper screenState>
+        <Form
+          key="invite-user-management"
+          initialValues={initialValuesForForm->JSON.Encode.object}
+          validate={values => UserManagementUtils.validateFormForRoles(values, groupsAdded)}
```

Comment:
validation is not working as expected.if we enter the role name first and select the permission button is still in disable state.